src/Pages/ProfilePage/ProfilePage.jsx

```diff
@@ -0,0 +1,165 @@
+  const { userState } = useAuthContext();
+  const { userFollowing, userFollowers, posts, users, dispatch } =
+    useReducerContext();
+  const { username } = useParams();
+    const { followUser, unfollowUser } = useUserContext();
+
+  useEffect(() => {
+    (async function () {
+      try {
+        const response = await axios.get("/api/posts");
+        if (response.status === 200) {
+          dispatch({ type: FETCH_POSTS, payload: response.data.posts });
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    })();
+  }, []);
+
+  return username === userState.username ? (
+    <div className={styles.profile_container}>
+      {modal && (
+        <EditProfileModal
+          firstName={userState.firstName}
+          bio={userState.bio}
+          profilePhoto={userState.profilePhoto}
+          coverPhoto={userState.coverPhoto}
+          setModal={setModal}
+        />
+      )}
+      {userState.coverPhoto ? (
+        <img
+          className={styles.profile_backdrop}
+          src={userState.coverPhoto}
+          alt=""
+        />
+      ) : (
+        <div className={styles.no_img}></div>
+      )}
+      <div className={styles.user_info}>
+        <div className={styles.user_img}>
+          <img
+            src={
+              userState.profilePhoto
+                ? userState.profilePhoto
+                : "https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png"
+            }
+            alt=""
+          />
+          <button onClick={() => setModal(true)}>Edit profile</button>
+        </div>
+        <div className={styles.user_data}>
+          <h1>{userState.firstName}</h1>
+          <h3>@{userState.username}</h3>
+          <p>{userState.bio}</p>
+          <div className={styles.follower_status}>
+            <p>
+              {userFollowing.length !== 0 ? userFollowing.length : 0} Following
+            </p>
+            <p>
+              {userFollowers.length !== 0 ? userFollowers.length : 0} Followers
+            </p>
+          </div>
+        </div>
+        <div className={styles.post_show_container}>
+          <p className={styles.post_header}>Posts</p>
+          <UserPosts
+            posts={posts.filter((post) => post.username === userState.username)}
+          />
+        </div>
+      </div>
+    </div>
+  ) : (
+    users
+      .filter((user) => user.username === username)
+      .map((user) => {
+        return (
+          <div key={user._id} className={styles.profile_container}>
+            {user.coverPhoto ? (
+              <img
+                className={styles.profile_backdrop}
+                src={user.coverPhoto}
+                alt=""
+              />
+            ) : (
+              <div className={styles.no_img}></div>
+            )}
+            <div className={styles.user_info}>
+              <div className={styles.user_img}>
+                <img
+                  src={
+                    user.profilePhoto
```

Comment:
suggestion - Nothing wrong with this approach but have you tried using nullish coalescing operator for this case - `user.profilePhoto ?? _link.png`.

Reply:
Will check it out